test/framework/clusterresourceset_helpers.go

```diff
@@ -152,3 +154,14 @@ func WaitForClusterResourceSetToApplyResources(ctx context.Context, input WaitFo
 	}, intervals...).Should(BeTrue())
 
 }
+
+func isResourceApplied(binding *addonsv1.ResourceSetBinding, resourceRef addonsv1.ResourceRef) bool {
```

Comment:
This could be a `Has()` method on `ResourceSetBinding.Resources`

Reply:
It is not a Has method since we check if it is applied here. We could divide this into 2 methods: getResource() plus isApplied(), but since there is no use for getter anywhere else, just added isApplied().

Reply:
Can this be removed in favor or `IsResourceApplied/IsApplied`? Looks it's the same code